examples/deck/example.js

```diff
@@ -133,6 +133,13 @@ export default class Example extends Component<
     window.addEventListener('resize', this._resize);
   }
 
+  componentDidUpdate(prevProps, prevState) {
+    if (this.state.mode === 'translate' && this.state.mode !== prevState.mode) {
+      /* eslint-disable-next-line react/no-did-update-set-state */
+      this.setState({ modeConfig: { enablePolygonSnapping: false, snapStrength: 0.2 } });
```

Comment:
This is required for snapping to work since the enable polygon snapping checkbox is associated with translate mode. There is a specific edge case that will case unsnapping to break.

The edge case is this: snap two features together, select a different mode, then select translate mode again. Leave the enable polygon snapping unchecked and try to unsnap the two polygons. Note that this cannot be done.

The problem is that without snapStrength specified through modeConfig, the code cannot determine at what mouse move threshold unsnapping should occur.

One thing I can do is to make the snapping related controls universal, where they will persist through mode changes. What do you think?

Reply:
I'm not seeing the need for an "unsnap" function. I think the logic could be more stateless. It can just "snap" if `pickObject` returns a gray edit handle and will do normal translate if not.

Something like this:

```js
if (modeConfig.snappingEnabled) {
  const picks = pickObjects({radius: modeConfig.snapPixels});
  // snapSourceEditHandle would be the purple handle being dragged
  const snapSourceEditHandle = getEditHandleBeingDragged(picks);
  // snapToEditHandle would be the gray handle close to the cursor or null if nothing close to the cursor
  const snapToEditHandle = getEditHandleOfNonSelectedFeature(picks);
  if (snapToEditHandle) {
    // perform snapping
    const startPoint = getCoordinateFromEditHandle(snapSourceEditHandle);
    const endPoint = getCoordinateFromEditHandle(snapToEditHandle);
    editAction = this.getTranslateAction(
        startPoint,
        endPoint,
        'translating'
      )
  }
}

if (!editAction) {
  // translate normally
 editAction = this.getTranslateAction(
        event.pointerDownGroundCoords,
        event.groundCoords,
        'translating'
      )
}
```